src/api/GoPayNotificationHandler.php

```diff
@@ -54,8 +60,20 @@ public function handle(ApiAuthorizationInterface $authorization)
         }
         $params = $paramsProcessor->getValues();
 
+        $gopayMeta = $this->gopayPaymentsRepository->getTable()
+            ->where(['transaction_reference' => $params['id']])
+            ->limit(1)->fetch();
+        if (!$gopayMeta) {
+            Debugger::log('Payment with transaction reference ' . $params['id'] . ' not found.', Debugger::EXCEPTION);
+            $response = new JsonResponse(['status' => 'ok']);
+            $response->setHttpCode(Response::S200_OK);
+            return $response;
+        }
+
+        $payment = $gopayMeta['payment'];
         try {
-            $result = $this->gopay->notification($params['id'], $params['parent_id'] ?? null);
+            $gateway = $this->gatewayFactory->getGateway($payment['payment_gateway']['code']);
+            $result = $gateway->notification($payment, $params['id'], $params['parent_id'] ?? null);
```

Comment:
So to make sure I (or anybody else reading this) understand correctly:

- You extended the implementation with another (internal) `GopayTrial` gateway.
- We need the `$payment` here in `NotificationHandler` so we can extract correct `$gateway` (`GopayRecurrent` or `GopayTrial`)
- We then call `$gateway->notification()` which is now moved to the `BaseGoPay` class and shared for all Gopay gateways.

The scenario you're trying to solve is:

- The notification can come for either `GopayRecurrent`, or your internal `GopayTrial`.
- You want both of them to be handled same way (that's why you moved notification to the `BaseGoPay`)

Correct?

Reply:
yes, basically I needed a way to handle notifications differently in different gopay gateways..